Source/Models/JSONDeserializer.swift

```diff
@@ -0,0 +1,30 @@
+import Argo
+import Result
+
+struct JSONDeserializer: Deserializer {
+  func deserialize<T : Request>(request: T)(data: NSData?) -> Result<T.Decoder.Representation, SwishError> {
```

Comment:
Implicit currying syntax is being removed. We should probably update this to use explicit currying.